docs/source/caching/cache-key-resolution.mdx

```diff
@@ -8,181 +8,66 @@ The normalized cache computes a **cache key** for each object that is stored in
-    return try? CacheKeyInfo(jsonValue: object["id"])
-  }
-}
+```graphql title="typePolicies.graphqls"
+extend type Author @typePolicy(keyFields: "firstName lastName")
 ```
 
-### Specifying cache IDs by `Object` type
-
-If you would like to specify **cache IDs** differently for different types of objects, you can use the `type` parameter.
-
-For an object of the type `Dog` with a unique key represented by an `id` field, you may implement cache key resolution as:
-
-```swift title="SchemaConfiguration.swift"
-public enum SchemaConfiguration: ApolloAPI.SchemaConfiguration {
-  static func cacheKeyInfo(for type: Object, object: ObjectData) -> CacheKeyInfo? {
-    switch type {
-    case Objects.Dog:
-      return try? CacheKeyInfo(jsonValue: object["id"])
-
-      default:
-      return nil
-    }
-  }
-}
-```
-
-### Specifying cache IDs by abstract types
-
-If object types sharing the same `interface` or `union` in your schema have the same cache key resolution strategy, you can resolve the key based on those abstract types.
+The resolved cache key will include all declared key fields separated by the `+` character. In this case, an `Author` object with a `firstName` of `"Ray"` and a last name of `"Bradbury"` would resolve to have a cache key of `Author:Ray+Bradbury`.
 
-The generated schema metadata includes `Interfaces` and `Unions` types that contain a list of all the abstract types used in your GraphQL schema.
+### Type policies on interface types
 
-For example, for a schema with `Dog` and `Cat` types that implement `interface Pet`, you may implement cache key resolution as:
+It is also possible to use `@typePolicy` on _interfaces_. This specifies a default set of key fields for all types that implement that interface.
 
-```swift title="SchemaConfiguration.swift"
-public enum SchemaConfiguration: ApolloAPI.SchemaConfiguration {
-  static func cacheKeyInfo(for type: Object, object: ObjectData) -> CacheKeyInfo? {
-    if type.implements(Interfaces.Pet) {
-      return try? CacheKeyInfo(jsonValue: object["id"])
-    }
-
-    return nil
-  }
-}
-```
-
-To instead configure cache key resolution based on a `union` type, use the union's [`possibleTypes`](https://www.apollographql.com/docs/ios/docc/documentation/apolloapi/union/possibletypes) property.
-
-```swift title="SchemaConfiguration.swift"
-public enum SchemaConfiguration: ApolloAPI.SchemaConfiguration {
-  static func cacheKeyInfo(for type: Object, object: ObjectData) -> CacheKeyInfo? {
-    if Unions.ClassroomPets.possibleTypes.contains(type) {
-      return try? CacheKeyInfo(jsonValue: object["id"])
-    }
-
-    return nil
-  }
-}
+```graphql title="extra.graphqls"
+extend interface Node @typePolicy(keyFields: "id")
 ```
 
-### Grouping cached objects with `uniqueKeyGroup`
+> **Note:** If an object type has a type policy of its own, it will take precedence and the type policy on an interface it implements will not be applied.
```

Comment:
The iOS Codegen will not do this, but will display a validation error instead.

Reply:
Thanks for the review! I misremembered this. I'll update the docs now.